packages/headless/src/controllers/commerce/search-box/headless-search-box.test.ts

```diff
@@ -0,0 +1,272 @@
+    const options: SearchBoxOptions = {
+      id,
+      highlightOptions: {
+        notMatchDelimiters: {
+          open: '<a>',
+          close: '<a>',
+        },
+        correctionDelimiters: {
+          open: '<i>',
+          close: '<i>',
+        },
+      },
+    };
+
+    props = {
+      options,
+    };
+
+    initState();
+    initController();
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  function initState() {
+    state = buildMockCommerceState();
+    state.querySet[id] = 'query';
+    state.querySuggest[id] = buildMockQuerySuggest({
+      id,
+      completions: [
+        {
+          expression: 'a',
+          score: 0,
+          executableConfidence: 0,
+          highlighted: '[hi]{light}(ed)',
+        },
+      ],
+    });
+  }
+
+  function initController() {
+    engine = buildMockCommerceEngine({state});
+    searchBox = buildSearchBox(engine, props);
+  }
+
+  it('it adds the correct reducers to engine', () => {
+    expect(engine.addReducers).toHaveBeenCalledWith({
+      commerceQuery,
+      querySuggest,
+      configuration,
+      querySet,
+      commerceSearch,
+    });
+  });
+
+  describe('validating options', () => {
+    it(`when passing an invalid id as option
+      creating the controller should throw`, () => {
+      props.options!.id = 1 as unknown as string;
+      expect(() => initController()).toThrow(
+        'Check the options of buildSearchBox'
+      );
+    });
+
+    it(`when passing an invalid highlightOptions as option
+      creating the controller should throw`, () => {
+      props.options!.highlightOptions = {
+        notMatchDelimiters: {
+          open: 1 as unknown as string,
+          close: 2 as unknown as string,
+        },
+      };
+      expect(() => initController()).toThrow();
+    });
+  });
+
+  it('should return the right state', () => {
+    expect(searchBox.state).toEqual({
+      value: state.querySet[id],
+      suggestions: state.querySuggest[id]!.completions.map((completion) => ({
+        highlightedValue: '<a>hi<a>light<i>ed<i>',
+        rawValue: completion.expression,
+      })),
+      isLoading: false,
+      isLoadingSuggestions: false,
+    });
+  });
+
+  it('should dispatch a registerQuerySetQuery action at initialization', () => {
+    expect(engine.actions).toContainEqual(
+      registerQuerySetQuery({id, query: state.commerceQuery.query!})
+    );
+  });
+
+  it('should dispatch a registerQuerySuggest action at initialization', () => {
+    expect(engine.actions).toContainEqual(
+      registerQuerySuggest({
+        id,
+      })
+    );
+  });
+
+  describe('when calling updateText', () => {
+    it('updates the search box query in the querySet', () => {
+      const text = 'query';
+      searchBox.updateText(text);
+
+      const action = updateQuerySetQuery({id, query: text});
+      expect(engine.actions).toContainEqual(action);
+    });
+
+    it('should call the showSuggestions method', () => {
```

Comment:
Actually, since `updateText` calls `showSuggestions`, it makes more sense to me that we test that this method is called. We don't care about what `showSuggestions` does!